packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -22,11 +22,7 @@ typedef ShutdownHook = FutureOr<dynamic> Function();
 // for more details.
 
 abstract class ShutdownHooks {
-  factory ShutdownHooks({
-    required Logger logger,
-  }) => _DefaultShutdownHooks(
-    logger: logger,
-  );
+  factory ShutdownHooks() => _DefaultShutdownHooks();
```

Comment:
Now that we construct this class before the context_runner, delay passing the Logger until `runShutdownHooks` is called.